pkg/controller/operators/catalog/operator.go

```diff
@@ -976,22 +976,28 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		// not-satisfiable error
 		if _, ok := err.(solver.NotSatisfiable); ok {
 			logger.WithError(err).Debug("resolution failed")
-			updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
+			subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
+			_, updateErr := o.updateSubscriptionStatuses(subs)
 			if updateErr != nil {
 				logger.WithError(updateErr).Debug("failed to update subs conditions")
+				return updateErr
 			}
 			return nil
 		}
-		updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ErrorPreventedResolution", err.Error(), true)
+		subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ErrorPreventedResolution", err.Error(), true)
+		_, updateErr := o.updateSubscriptionStatuses(subs)
 		if updateErr != nil {
 			logger.WithError(updateErr).Debug("failed to update subs conditions")
+			return updateErr
 		}
 		return err
 	} else {
```

Comment:
The "else" is unnecessary if the function is guaranteed to return on error, isn't it?